spec/models/service_template_transformation_plan_spec.rb

```diff
@@ -354,12 +378,25 @@
       expect(service_template.transformation_mapping).to eq(transformation_mapping)
       expect(service_template.vm_resources.collect(&:resource)).to match_array([vm1, vm2, vm3])
       expect(service_template.vm_resources.collect(&:status)).to eq([ServiceResource::STATUS_QUEUED, ServiceResource::STATUS_QUEUED, ServiceResource::STATUS_QUEUED])
-      expect(service_template.vm_resources.find_by(:resource_id => vm1.id).options)
-        .to eq("pre_ansible_playbook_service_template_id" => apst.id, "osp_security_group_id" => security_group1.id, "osp_flavor_id" => flavor1.id)
-      expect(service_template.vm_resources.find_by(:resource_id => vm2.id).options)
-        .to eq("pre_ansible_playbook_service_template_id" => apst.id, "post_ansible_playbook_service_template_id" => apst.id, "osp_security_group_id" => security_group1.id, "osp_flavor_id" => flavor1.id)
-      expect(service_template.vm_resources.find_by(:resource_id => vm3.id).options)
-        .to eq("pre_ansible_playbook_service_template_id" => apst.id, "post_ansible_playbook_service_template_id" => apst.id, "osp_security_group_id" => security_group2.id, "osp_flavor_id" => flavor2.id)
+
+      stub_const("VirtualMachine", Class.new(ManageIQ::Providers::InfraManager::Vm) do
+        supports :warm_migrate do
+          unsupported_reason_add(:warm_migrate, _('Warm migratiobn can not migrate a VM with snapshots')) if snapshots.present?
+        end
+      end)
+
+      vm1_options = service_template.vm_resources.find_by(:resource_id => vm1.id).options
+      vm1_options["warm_migration_compatible"] = VirtualMachine.new(:snapshots => vm1.snapshots).supports_warm_migrate?
```

Comment:
Instead of using `vm1.snapshots` could this warm-migration compatible class _be_ vm1

Reply:
I did start with vm1 but using `VirtualMachine.new(:vm => vm1).supports_warm_migrate?` or `VirtualMachine.new(:vm_or_template => vm1).supports_warm_migrate?` would complain that `:vm`, `vm_or_template` were UNKNOWN.  Using `:snapshot` was the only I could get it work.

Reply:
No, instead of `let(:vm1) { FactoryBot.create(:vm_or_template) }` you do `let(:vm1) { VirtualMachine.new }`